src/main/java/ui/GUIController.java

```diff
@@ -102,11 +125,54 @@ public void panelFilterExpressionChanged(FilterPanel changedPanel) {
                 .collect(Collectors.toList());
     }
 
-    private void updateAPIBox(UpdateRateLimitsEvent e) {
-        Platform.runLater(() -> apiBox.setText(String.format("%s/%s",
-                                                             e.remainingRequests,
-                                                             Utility.minutesFromNow(e.nextRefreshInMillisecs)))
-        );
+    /**
+     * Updates UI components using the API rate limits information.
+     * @param e The current API rate limits
+     */
+    private void updateRateLimits(RateLimitsUpdatedEvent e) {
+        updateAPIBox(e.remainingRequests, e.nextRefreshInMillisecs);
+    }
+
+    /**
+     * Updates the period of the refresh timer for synchronization of the data store.
+     * @param e The current API rate limits for calculation of the refresh rate.
+     */
+    private void updateSyncRefreshRate(RefreshTimerTriggeredEvent e) {
+        apiCallsUsedInPreviousRefresh = computePreviousRemainingApiRequests(e.remainingRequests);
+        refreshDurationInMinutes = RefreshTimer.computeRefreshTimerPeriod(e.remainingRequests,
+                                                                    Utility.minutesFromNow(e.nextRefreshInMillisecs),
+                                                                    apiCallsUsedInPreviousRefresh,
+                                                                    RefreshTimer.API_QUOTA_BUFFER,
+                                                                    RefreshTimer.DEFAULT_REFRESH_PERIOD_IN_MINS);
+        ui.refreshTimer.changeRefreshPeriod((int) refreshDurationInMinutes);
+        logger.info("Refresh period updated to " + refreshDurationInMinutes
+                    + "mins with apiCalls used in previous refresh cycle is " + apiCallsUsedInPreviousRefresh
+                    + ", current api quota is " + e.remainingRequests + " and next quota top-up in "
+                    + Utility.minutesFromNow(e.nextRefreshInMillisecs) + "mins.");
+
+    }
+
+    private int computePreviousRemainingApiRequests(int remainingRequests) {
```

Comment:
Something doesn't seem to be right here. Write a header comment so that we can compare the intent with the implementation.